### ISSUE TITLE
Provide default Grafana dashboards for the produced metrics

### ISSUE THREAD

**Issue by Inithron** [quarkusio/quarkus#40933]
### Description

Hi,
I am using Quarkus with the micrometer and some other extensions. Everything works fine but for example the Kafka extension creates a huge amount of metrics. I am new to this topic so I have problems to figure out which of the metrics are really important or valuable. I have found a dashboard visualizing some basic metrics: https://grafana.com/grafana/dashboards/14370-jvm-quarkus-micrometer-metrics which I like very much.

### Implementation ideas

My proposal now is to provide similar dashboards for the other extensions: https://quarkus.io/guides/telemetry-micrometer#quarkus-extensions-using-micrometer. This would help everybody visualizing the metrics without spending days in getting familiar with Grafana and creating their own dashboards.

After the creation of this enhancement I have seen that there is a [Grafana Dashboard Generator](https://github.com/jamesfalkner/microprofile-grafana). Unfortunately the generator is not really easy to install/use and the last commit was 4 years ago. So I am not sure if it still works but it gives me another idea: Instead of creating manually Grafana dashboards the dashboard generator could be added to the Quarkus Dev Services or into an own Quarkus extension.

**Comment by geoand** [comment]
cc @alesj @brunobat

**Comment by alesj** [comment]
We've just added this:
* https://quarkus.io/version/main/guides/observability-devservices

If there is some easy way to pre-add custom / specific dashboard(s) via Grafana configuration, this would be the way to do it.
e.g. adding the right config into the container
* https://github.com/quarkusio/quarkus/blob/main/extensions/observability-devservices/testcontainers/src/main/java/io/quarkus/observability/testcontainers/GrafanaContainer.java

**Comment by melloware** [comment]
@Inithron i had just asked this same thing on Zulip.  I have that same Grafana dashboard as you and I think the goal is to have that in the LGTM stack out of the box.

**Comment by Inithron** [comment]
@alesj, @melloware I think there was a misunderstanding. My intention was not add the existing dashboard to the Dev Services. I would like to have one dashboard containing all metrics which are produced by the Quarkus extensions: https://quarkus.io/guides/telemetry-micrometer#quarkus-extensions-using-micrometer or one dashboard for each extension. The alternative would be to integrate the dashboard generator into the Dev Services to generate the dashboard(s) automatically.

**Comment by melloware** [comment]
@Inithron I see so ideally we have dashboards for all the things like Kafka and any other MicroMeter Metrics.  Those deashboards should default in the LGTM DevResource so they come out of the box.  People can then "export" those dashboards from Grafana to use in their production Grafana environment.

**Comment by Inithron** [comment]
Yes, that would be great. But first someone needs to create these dashboards.

**Comment by melloware** [comment]
Yep exactly what I asked on Zulip: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Observabilty.20Dev.20Services.20GRAFANA.20OTEL.20LGTM

**Comment by melloware** [comment]
it doesn't work but here is something of a prototype whree we need to add a Prometheus scraper to scrap `/q/metrics` to get the Micrometer data into dashboard.

```java
public class LgtmContainer extends GrafanaContainer<LgtmContainer, LgtmConfig> {
    protected static final String LGTM_NETWORK_ALIAS = "ltgm.testcontainer.docker";

    protected static final String PROMETHEUS_CONFIG = """
                global:
                  scrape_interval: 15s
                  evaluation_interval: 15s
                storage:
                  tsdb:
                    out_of_order_time_window: 10m
                scrape_configs:
                  - job_name: 'quarkus-micrometer'
                    metrics_path: '/q/metrics'
                    scrape_interval: 15s
                    static_configs:
                      - targets: ['ltgm.testcontainer.docker:8080']
            """;

    public LgtmContainer() {
        this(new LgtmConfigImpl());
    }

    public LgtmContainer(LgtmConfig config) {
        super(config);
        addExposedPorts(config.otlpPort());
        addFileToContainer(PROMETHEUS_CONFIG.getBytes(), "/otel-lgtm/prometheus.yaml");

    }
```

**Comment by melloware** [comment]
Ok I got it working. Next I need to figure out how to load the dashboard I want. But making good progress!

**Comment by alesj** [comment]
Do we need this scraping?
Since isn't OTLP supposed to push this into collector, and the this fwds to Prometheus -> Grafana?

**Comment by brunobat** [comment]
Correct @alesj. On the grafana devservice there is no scraping. It's all OTLP protocol because it includes an OTel Collector.
The dashboard is usually placed on a specific folder of Grafana, which we have no access to... We can always ask the Grafana folks. 
Another solution is to add the dashboard through API or create a dashboard on the Grafana dashboard registry.

**Comment by melloware** [comment]
@alesj I do not see the micrometer metrics available out of the box with your current LGTM stack.  I heard there is something called a "Micrometer Bridge" but i can't find any reference but I can confirm the JVM and Micrometer metrics are not flowing through to LGTM container.

**Comment by melloware** [comment]
@alesj @brunobat according to this document the Microemter stuff is disabled by default: https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v2.0.0

> Micrometer metric bridge has been disabled by default. You can enable it using OTEL_INSTRUMENTATION_MICROMETER_ENABLED=true or `-Dotel.instrumentation.micrometer.enabled=true`.

**Comment by brunobat** [comment]
> @alesj @brunobat according to this document the Micrometer stuff is disabled by default: https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v2.0.0
> 
> > Micrometer metric bridge has been disabled by default. You can enable it using OTEL_INSTRUMENTATION_MICROMETER_ENABLED=true or `-Dotel.instrumentation.micrometer.enabled=true`.

We don't support that yet.

**Comment by melloware** [comment]
@brunobat @alesj yes that is why i am working on my PR to have a kick button default Quarkus Dashboard.  I am going to keep going on it and submit a PR you guys can choose to do what you want with it.

**Comment by melloware** [comment]
![image](https://github.com/quarkusio/quarkus/assets/4399574/1cba9f98-36ec-4d5a-9f50-354aaf5fcd78)

**Comment by Inithron** [comment]
Hi @melloware,
I have checked #41264 and the screenshot above. But I can't see the metrics for Kafka or other extension. So is my assumption correct, that these metrics are not included in the dashboard?

**Comment by melloware** [comment]
Correct we decided to go with the simplest metrics possible to start with.

**Comment by Inithron** [comment]
I appreciate the work which was done so fare. But from a functional point of view this dashboard provides more or less the same information as the dashboard I have linked in this feature request. So the goal, to have dashboards for Kafka and other Quarkus extensions, is not fulfilled. @geoand can you please reopen this feature request?

**Comment by geoand** [comment]
I will leave that up to the folks that worked on it.

**Comment by melloware** [comment]
@Inithron PR is welcome.  I don't use Kafka so I won't be making a dashboard for it but you seem to have the need so building one and submitting a PR should be easy.

**Comment by Inithron** [comment]
@melloware you didn't answer the question. I created this feature request to get dashboards for the metrics which are produced by the Quarkus extension. But you created a dashboard wich exists already. So do you agree to reopen this feature request?

**Comment by melloware** [comment]
Well I will leave this up to @brunobat where he stated that there are N+ extensions that expose metrics and no one uses every extension so one dashboard wouldn't make sense. So he suggested a separate dashboard per extension in your case Kafka. 

So I am not sure one generic ticket of dashboard for all exposed metrics makes sense but I will let @brunobat chine in.

**Comment by Inithron** [comment]
One generic dashboard makes definitely no sense. But from the beginning on I was talking about dashboards, plural. And if you scroll up and read the initial request again, you will see this is the same what I have suggested:
`My proposal now is to provide similar dashboards for the other extensions: https://quarkus.io/guides/telemetry-micrometer#quarkus-extensions-using-micrometer.`

**Comment by brunobat** [comment]
@Inithron thanks for the feedback.
As @melloware mentioned this PR was to provide the minimal setup to have Dashboards up and running.
This issue was created with follow up improvements: https://github.com/quarkusio/quarkus/issues/43599
What you are requesting, a Kafka specific dashboard makes sense and there are multiple ways to do it:
- Have a section on the dashboard. If we do this for all extensions, the dashboard will be cluttered, but that's not a bib problem because this dashboard was intended to be a template that users could use to create their own dashbords... Maybe we can provide a config to load your own dashboard in the devservice?
-  Have extension specific dashboards, or get dashboards that provide insights to particular types of apps. A kafka based app, probably doesn't need the HTTP related dashboard components. We can allow people to submit domain specific dashbords that make sense portions of the user base.
- Automatic generation of the dashboard based on automatic generation. This looks appealing but I belied it's out of scope due to it's complexity and the size of our backlog. However, contributions are welcome. 

In conclusion. There are many things that can be done in the dashboard area and we need to be pragmatic. Please consider these dashboards a starting point to your own dashboards that eventually can be shared through Quarkus. 
Please create a discussion about this because this issue is closed.

**Comment by Inithron** [comment]
Hi @brunobat,
For the pull request this is totally ok. But in this feature request I requested default dashboards for all Quarkus extensions. Not only for Kafka. This was only an example because the Kafka extension creates a lot of metrics. But now no new dashboard was created. You only recreated the already existing one and added it to the dev services. I like this approach and I appreciate the work you did. But this was something completely different than I have requested. That's why I think this feature request should be reopened, or do you see it different?

**Comment by brunobat** [comment]
Sorry @Inithron. In my opinion the issue was too vage. We need more concrete steps than "provide similar dashboards for the other extensions: https://quarkus.io/guides/telemetry-micrometer#quarkus-extensions-using-micrometer."
And also how where to implement ir.

We assumed, Grafana LGTM dev service and provide the basic infra to iterate on the dashboard.
I understand that support metrics for other extensions is needed but we cannot boil the ocean here. 

I see this issue as a starting point. Before this, there were not dashboards on the devservices. This is the feature that will allow us to expand them, test them and only then, publish on the Grafana registry.

I'm open to open a new issue, an umbrella issue, where we can detail which metrics and in which extensions we will work next, but the list needs to be concrete in order to prioritise those sub issues.

**Comment by brunobat** [comment]
I think you should create the discussion first.

**Comment by Inithron** [comment]
I just wanted to create the discussion but I am not sure in which category I should create it: In Community, Design Discussions, Q&A or should I create a Poll?

**Comment by brunobat** [comment]
> I just wanted to create the discussion but I am not sure in which category I should create it: In Community, Design Discussions, Q&A or should I create a Poll?

Design discussion should be fine
